src/components/account/ResetPassword.test.tsx

```diff
@@ -0,0 +1,48 @@
+import UserService from '@services/user.service'
+import {cleanup, render, screen, waitFor} from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import {REQUEST_RESET_PASSWORD_MESSAGE} from '__test_utils/test_server/userHandlers'
+import {createWrapper} from '__test_utils/utils'
+import {ResetPassword} from './ResetPassword'
+
+export const setUp = () => {
+  const user = userEvent.setup()
+  const spyOnRequestResetPassword = jest.spyOn(UserService, 'requestResetPassword')
+  const element = render(<ResetPassword />, {wrapper: createWrapper()})
+
+  const emailField = screen.getByLabelText('Email')
+  const submitButton = screen.getByRole('button', {name: /reset my password/i})
+
+  return {user, spyOnRequestResetPassword, element, emailField, submitButton}
+}
+
+afterEach(() => {
+  jest.restoreAllMocks()
+  cleanup()
+})
+
+describe.only('ResetPassword', () => {
+  test('should not be able to request password reset when email is empty', async () => {
+    const {emailField, submitButton, spyOnRequestResetPassword} = setUp()
+    expect(emailField).toHaveTextContent('')
+    expect(submitButton).toBeDisabled()
+    expect(spyOnRequestResetPassword).toHaveBeenCalledTimes(0)
+  })
+
+  test('should show success banner when password reset request is sent', async () => {
+    const {user, emailField, submitButton, spyOnRequestResetPassword} = setUp()
+    const testEmail = 'test@fake.com'
+
+    await waitFor(async () => {
+      await user.type(emailField, testEmail)
+      await user.click(submitButton)
+    })
```

Comment:
This `waitFor` is probably not necessary since it will just re-run if either of these actions throw an error. If either of these actions _do_ throw, then you should probably `await findBy...` the element that hasn't appeared yet, instead of wrapping the actions.

Reply:
Good call! I moved the actions outside of the `waitFor`. Since the `emailField` was already returned via a `getBy` query in the `setUp`, I then `waitFor` the `emailField` to be updated with the test email address -- which I think adds up to a `findBy`, since `findBy` is a combination of `getBy*` queries and `waitFor`.